src/modules/keyboardmanager/KeyboardManagerEngineLibrary/trace.cpp

```diff
@@ -82,6 +83,181 @@ void Trace::ShortcutRemapInvoked(bool isShortcutToShortcut, bool isAppSpecific)
     }
 }
 
+// Function to return a human readable string for the shortcut
+std::wstring GetShortcutHumanReadableString(Shortcut const & shortcut, LayoutMap& keyboardMap)
+{
+    std::wstring humanReadableShortcut = L"";
+    if (shortcut.winKey != ModifierKey::Disabled)
+    {
+        humanReadableShortcut += keyboardMap.GetKeyName(shortcut.GetWinKey(ModifierKey::Both)) + L" + ";
+    }
+    if (shortcut.ctrlKey != ModifierKey::Disabled)
+    {
+        humanReadableShortcut += keyboardMap.GetKeyName(shortcut.GetCtrlKey()) + L" + ";
+    }
+    if (shortcut.altKey != ModifierKey::Disabled)
+    {
+        humanReadableShortcut += keyboardMap.GetKeyName(shortcut.GetAltKey()) + L" + ";
+    }
+    if (shortcut.shiftKey != ModifierKey::Disabled)
+    {
+        humanReadableShortcut += keyboardMap.GetKeyName(shortcut.GetShiftKey()) + L" + ";
+    }
+    if (shortcut.actionKey != NULL)
+    {
+        humanReadableShortcut += keyboardMap.GetKeyName(shortcut.actionKey);
+    }
+    return humanReadableShortcut;
+}
+
+
+// Log the current remappings of key and shortcuts when keyboard manager engine loads the settings.
+void Trace::SendKeyAndShortcutRemapLoadedConfiguration(State& remappings) noexcept
+{
+    LayoutMap keyboardMap;
+    for (auto const& keyRemap : remappings.singleKeyReMap)
+    {
+        if (keyRemap.second.index() == 0)
```

Comment:
comment would be nice here for == 0

Reply:
not scope of this PR, but enum should be introduced

Reply:
I've added some comments. Thank you!